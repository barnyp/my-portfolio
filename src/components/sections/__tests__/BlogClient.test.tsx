import React from 'react';
import { render, screen } from '@testing-library/react';
import BlogClient from '../blog-client';
import '@testing-library/jest-dom';

// Mock framer-motion globally via jest.setup, so no additional mocks here.

const mockPosts = [
  {
    title: 'Post One',
    date: '2025-01-01',
    slug: 'post-one',
    author: 'Test Author',
    coverImage: '/img/test-1.jpg',
    excerpt: 'Excerpt 1',
  },
  {
    title: 'Post Two',
    date: '2025-02-01',
    slug: 'post-two',
    author: 'Test Author',
    coverImage: '/img/test-2.jpg',
    excerpt: 'Excerpt 2',
  },
];

describe('BlogClient Component', () => {
  it('renders a card for each post', () => {
    render(<BlogClient posts={mockPosts} />);
    mockPosts.forEach(post => {
      expect(screen.getByText(post.title)).toBeInTheDocument();
    });
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<BlogClient posts={mockPosts} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
