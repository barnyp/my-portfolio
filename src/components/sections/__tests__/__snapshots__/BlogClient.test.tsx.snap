// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BlogClient Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="container py-20"
  >
    <div
      class="text-center"
      initial="[object Object]"
      transition="[object Object]"
      viewport="[object Object]"
      whileinview="[object Object]"
    >
      <h2
        class="text-3xl font-bold"
      >
        Blog
      </h2>
      <h3
        class="text-xl font-semibold text-primary"
      >
        Latest From My Blog
      </h3>
    </div>
    <div
      class="grid md:grid-cols-2 lg:grid-cols-3 gap-10 mt-10"
      initial="hidden"
      variants="[object Object]"
      viewport="[object Object]"
      whileinview="visible"
    >
      <div
        class="bg-card border rounded-lg overflow-hidden shadow-lg hover:shadow-xl transition-shadow duration-300"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <div
          class="relative overflow-hidden"
        >
          <img
            alt="Post One"
            class="w-full h-48 object-cover transition-transform duration-300 hover:scale-105"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=%2Fimg%2Ftest-1.jpg&w=1080&q=75"
            srcset="/_next/image?url=%2Fimg%2Ftest-1.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg%2Ftest-1.jpg&w=1080&q=75 2x"
            style="color: transparent;"
            width="500"
          />
          <div
            class="absolute top-4 left-4"
          >
            <span
              class="bg-primary text-primary-foreground px-2 py-1 rounded-full text-xs font-medium"
            >
              #1
            </span>
          </div>
        </div>
        <div
          class="p-6"
        >
          <div
            class="flex items-center gap-4 text-sm text-muted-foreground mb-3"
          >
            <span
              class="flex items-center gap-1"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-calendar w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 2v4"
                />
                <path
                  d="M16 2v4"
                />
                <rect
                  height="18"
                  rx="2"
                  width="18"
                  x="3"
                  y="4"
                />
                <path
                  d="M3 10h18"
                />
              </svg>
              2025-01-01
            </span>
            <span
              class="flex items-center gap-1"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-clock w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <polyline
                  points="12 6 12 12 16 14"
                />
              </svg>
              2 min read
            </span>
          </div>
          <h4
            class="font-bold text-xl mb-2 line-clamp-2 hover:text-primary transition-colors"
          >
            Post One
          </h4>
          <p
            class="text-muted-foreground mb-4 line-clamp-3"
          >
            Excerpt 1
          </p>
          <a
            href="/blog/post-one"
          >
            <button
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-full group"
            >
              Read More
              <div
                class="ml-2"
                initial="[object Object]"
                transition="[object Object]"
                whilehover="[object Object]"
              >
                →
              </div>
            </button>
          </a>
        </div>
      </div>
      <div
        class="bg-card border rounded-lg overflow-hidden shadow-lg hover:shadow-xl transition-shadow duration-300"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <div
          class="relative overflow-hidden"
        >
          <img
            alt="Post Two"
            class="w-full h-48 object-cover transition-transform duration-300 hover:scale-105"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=%2Fimg%2Ftest-2.jpg&w=1080&q=75"
            srcset="/_next/image?url=%2Fimg%2Ftest-2.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg%2Ftest-2.jpg&w=1080&q=75 2x"
            style="color: transparent;"
            width="500"
          />
          <div
            class="absolute top-4 left-4"
          >
            <span
              class="bg-primary text-primary-foreground px-2 py-1 rounded-full text-xs font-medium"
            >
              #2
            </span>
          </div>
        </div>
        <div
          class="p-6"
        >
          <div
            class="flex items-center gap-4 text-sm text-muted-foreground mb-3"
          >
            <span
              class="flex items-center gap-1"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-calendar w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 2v4"
                />
                <path
                  d="M16 2v4"
                />
                <rect
                  height="18"
                  rx="2"
                  width="18"
                  x="3"
                  y="4"
                />
                <path
                  d="M3 10h18"
                />
              </svg>
              2025-02-01
            </span>
            <span
              class="flex items-center gap-1"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-clock w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <polyline
                  points="12 6 12 12 16 14"
                />
              </svg>
              2 min read
            </span>
          </div>
          <h4
            class="font-bold text-xl mb-2 line-clamp-2 hover:text-primary transition-colors"
          >
            Post Two
          </h4>
          <p
            class="text-muted-foreground mb-4 line-clamp-3"
          >
            Excerpt 2
          </p>
          <a
            href="/blog/post-two"
          >
            <button
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-full group"
            >
              Read More
              <div
                class="ml-2"
                initial="[object Object]"
                transition="[object Object]"
                whilehover="[object Object]"
              >
                →
              </div>
            </button>
          </a>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
